src/git.rs

```diff
@@ -193,7 +186,21 @@ impl Git for LiveGit {
         for &exclusion in exclusions.iter().filter(|id| !id.is_zero()) {
             revwalk.hide(exclusion)?;
         }
-        revwalk.hide_head()?;
+        for mainline in &self.config.mainlines {
+            if mainline == "HEAD" {
+                revwalk.hide_head()?;
+            } else if mainline.contains(|c| c == '?' || c == '*' || c == '[') {
+                revwalk.hide_glob(&format!("refs/heads/{}", mainline))?;
+            } else {
+                match revwalk.hide_ref(&format!("refs/heads/{}", mainline)) {
```

Comment:
What was the problem with not just using `hide_glob` here again?

Reply:
If you pass `hide_glob` something that isn't a glob, it expands it to `thing-you-passed/*`
I'm not entirely sure why, but this is usually coupled with it complaining that the `thing-you-passed/`directory doesn't exist, and not matching the `thing-you-passed` branch.